documentation/writing_tests.md

```diff
@@ -14,6 +17,77 @@
 
 ## Recommended tools
 
-## Structure of tests
+## Test folder structure
 
-## Individual tests
\ No newline at end of file
+Tests for the Azure SDK for JavaScript and TypeScript should be all executed using the NPM command `test`, they should be stored in the `test` folder, they must be written in TypeScript, in files that will end in `.spec.ts`. Our [recommended tools](#recommended-tools) should provide the test framework and proper ways of compiling and bundling our tests correctly. They should be able to run and be debugged on the environments we're targeting, which are currently only NodeJS and the browsers.
+
+To help developers specify the target environment they will be testing, we will specify these scripts in the `package.json`:
+
+```json
+  "test:browser": "karma start --single-run",
+  "test:node": "mocha -r esm --require source-map-support/register --reporter mocha-multi --reporter-options spec=-,mocha-junit-reporter=- --timeout 180000 --full-trace dist-esm/test/*.test.js",
```

Comment:
After merging either this or [this](https://github.com/sadasant/azure-sdk-for-js/pull/14/) PR, I should update these lines to be aligned with the contents on [this](https://github.com/sadasant/azure-sdk-for-js/pull/14/) PR.